scripts/ruby/lib/dataManager.rb

```diff
@@ -108,17 +109,38 @@ def batch_process_files
 
   def check_options
     # verify that everything's all good before moving to per-file level processing
-    if should_transform("es")
+    if should_transform?("es")
       assert_option("es_path")
       assert_option("es_index")
       assert_option("es_type")
     end
 
-    if should_transform("solr")
+    if should_transform?("solr")
       assert_option("solr_core")
       assert_option("solr_path")
     end
+  end
 
+  def check_schema
+    # if ES is requested and not transform only, then check the schema
+    # if there is no schema for this type, shut it all down
+    if should_transform?("es") && !@options["transform_only"]
+      url = "#{@options["es_path"]}/#{@options["es_index"]}/_mapping/#{@options["es_type"]}?pretty"
+      msg = "\nSchema url: #{url}\n"
+      begin
+        res = RestClient.get(url)
+        body = res.body
+      rescue => e
+        msg += e.to_s.red
+        raise msg
+      end
+      json = JSON.parse(body)
+      if json.nil? || json.empty?
+        msg += "No schema found for #{@options["es_type"]}\n".red
```

Comment:
Looks great! I'd just replace these `+=` concats with the `<<` syntax to be consistent and for efficiency: https://github.com/bbatsov/ruby-style-guide#concat-strings